### DISCUSSION TITLE
SQL snippet to grant role to user

### DISCUSSION THREAD

**Discussion by ahoward2** [supabase/supabase#3792] [Questions] [answered]
Hey everyone! :)

This is probably a super basic question but I've looked everywhere in the docs and the internet.

I want to simply grant a role to a specific user in the auth.users table.

Would it be something along the lines of:

grant role to user_id

**Comment by dshukertjr** [comment, marked as answer]
Hi @ahoward2!

Could I ask what the exact purpose of granting a role to a auth user?

In Supabase, you control what users can and cannot access with [row level security](https://supabase.io/docs/guides/auth/row-level-security), and you should generally never have to grant roles to individual users.

**Reply by ahoward2** [reply]
Thanks for the response @dshukertjr,

I was thinking about this the wrong way. For my application I want groups of users to have different access, but I think I should be distinguishing between application roles vs roles in supabase. 

For example if in my application I'm matching tutors with students, I might rather have a "tutors" table and "students" table that have respective auth.user.id foreign keys, then do a check in my application to see if an authenticated user exists in one of those tables, and render or redirect accordingly?

**Reply by dshukertjr** [reply]
@ahoward2 Yup, exactly! I also highly recommend watching [this](https://www.youtube.com/watch?v=Ow_Uzedfohk) video if you haven't yet to learn more about advanced row level security techniques that you can use inn your Supabase project!
